sdks/python/apache_beam/examples/inference/tensorrt_object_detection.py

```diff
@@ -0,0 +1,238 @@
+    filename, im_width, im_height = key
+    nums = prediction_result.inference[0]
+    boxes = prediction_result.inference[1]
+    scores = prediction_result.inference[2]
+    classes = prediction_result.inference[3]
+    detections = []
+    for i in range(int(nums[0])):
+        detections.append({
+                'ymin': str(boxes[i][0] * im_height),
+                'xmin': str(boxes[i][1] * im_width),
+                'ymax': str(boxes[i][2] * im_height),
+                'xmax': str(boxes[i][3] * im_width),
+                'score': str(scores[i]),
+                'class': COCO_OBJ_DET_CLASSES[int(classes[i])]
+            })
+    yield filename + ',' + str(detections)
+
+def parse_known_args(argv):
+  """Parses args for the workflow."""
+  parser = argparse.ArgumentParser()
+  parser.add_argument(
+      '--input',
+      dest='input',
+      required=True,
+      help='Path to the text file containing image names.')
+  parser.add_argument(
+      '--output',
+      dest='output',
+      required=True,
+      help='Path where to save output predictions.'
+      ' text file.')
+  parser.add_argument(
+      '--engine_path',
+      dest='engine_path',
+      required=True,
+      help='Path to the pre-built TFOD ssd_mobilenet_v2_320x320_coco17_tpu-8' 
+      'TensorRT engine.')
+  parser.add_argument(
+      '--images_dir',
+      default=None,
+      help='Path to the directory where images are stored.'
+      'Not required if image names in the input file have absolute path.')
+  return parser.parse_known_args(argv)
+
+
+def run(argv=None, save_main_session=True):
+  """
+  Args:
+    argv: Command line arguments defined for this example.
+  """
+  known_args, pipeline_args = parse_known_args(argv)
+  pipeline_options = PipelineOptions(pipeline_args)
+  pipeline_options.view_as(SetupOptions).save_main_session = save_main_session
+
+  engine_handler = KeyedModelHandler(
+      TensorRTEngineHandlerNumPy(
+          min_batch_size=1, max_batch_size=1, 
+        engine_path=known_args.engine_path))
+
+  with beam.Pipeline(options=pipeline_options) as p:
+    filename_value_pair = (
+        p
+        | 'ReadImageNames' >> beam.io.ReadFromText(
+            known_args.input, skip_header_lines=0)
+        | 'ReadImageData' >> beam.Map(
+            lambda image_name: read_image(
+                image_file_name=image_name, path_to_dir=known_args.images_dir))
+        | 'AttachImageSizeToKey' >> beam.Map(attach_im_size_to_key)
+        | 'PreprocessImages' >> beam.MapTuple(
+            lambda file_name, data: (file_name, preprocess_image(data))))
+    predictions = (
+        filename_value_pair
+        | 'PyTorchRunInference' >> RunInference(engine_handler)
+        | 'ProcessOutput' >> beam.ParDo(PostProcessor()))
+
+    if known_args.output:
+      predictions | "WriteOutputToGCS" >> beam.io.WriteToText(
+        known_args.output,
+        shard_name_template='',
+        append_trailing_newlines=True)
+
```

Comment:
@ryanthompson591 Great suggestion. However, I wanted to show you with example what will happen. In object detection you are predicting a bunch of things per detection: 4 box coordinates, score, class. Considering that there are 100 detections per image. For every image it will look like this:
![image](https://user-images.githubusercontent.com/101208641/178771528-b6de0658-5247-4731-b5cd-7308f2cbcc2f.png)

Do you think it is ok to print this to user in case no output file was specified? Let me know if I should implement this else clause or take it out.

Reply:
Yeah, looks like a little too much output to be human readable. 

I think what brought on my comment was that it seemed strange to run the pipeline with no output at all.

Some other things you could do:
```
else:
  LOG.warning('No output file specified. No output will be written')
```
or just make the output field mandatory.

It's up to you.

Reply:
I'll just keep it required. I was looking at [pytorch_image_classification.py ](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/inference/pytorch_image_classification.py) when designing it. For some reason output is actually [required](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/inference/pytorch_image_classification.py#L82), hence no idea why even make an [if statement](https://github.com/apache/beam/blob/master/sdks/python/apache_beam/examples/inference/pytorch_image_classification.py#L138).  I just took it for granted and missed this small logic mistake. I'll just take out if statement considering rest of the code and documentation requires presence of an output.

Reply:
@yeandy makes sense now, thanks!